external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -360,53 +391,95 @@ void outlineConvPartOps(Operation *convOp, ArrayRef<Operation *> secondOps,
   }
 
   // Erase the ops we outlined, which should be safe now.
-  for (auto &op : llvm::make_early_inc_range(llvm::reverse(secondOps))) {
+  for (auto &op : llvm::make_early_inc_range(llvm::reverse(trailingOps))) {
     if (op->use_empty()) {
       op->erase();
     }
   }
-  assert(convOp->use_empty() && "expected 'op' to have no uses");
-  convOp->erase();
-  for (auto &op : llvm::make_early_inc_range(frontOps)) {
+  assert(anchorOp->use_empty() && "expected 'op' to have no uses");
+  anchorOp->erase();
+  for (auto &op : llvm::make_early_inc_range(leadingOps)) {
     if (op->use_empty()) {
       op->erase();
     }
   }
 }
 
+} // namespace
+
+namespace mlir {
+namespace tosa {
+
+class PartitionConfig {
```

Comment:
New stuff.  A PartitionConfig has three predicates to define anchor, leading, and trailing ops, and a string attribute name.  No explicit "trailing-only" at this level;  instead, isLeadingOp() will be made to return false always.

Reply:
Please move this to a header file: mlir/Dialect/Tosa/Transforms/TosaPartition.h
so other users can derive from this class for customized partitioning.

Reply:
Yes.  I realised I'd forgotten that part of your original requests.